common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -3215,6 +3215,10 @@ public static enum ConfVars {
         "Time in seconds after which a compaction job will be declared failed and the\n" +
         "compaction re-queued."),
 
+    HIVE_COMPACTOR_WORKER_SLEEP_TIME("hive.compactor.worker.sleep.time", "10000ms",
```

Comment:
do we really need to expose this as a parameter?

Reply:
This config was added to remove the hardcoded sleep time in the code, also to avoid extensive sleep time while unit test are executed.

Reply:
do we really need to expose this complexity to the end users?

Reply:
The reason why these needs to be configurable is so that these values can be tweaked while testing as well, there are testcases like these https://github.com/apache/hive/blob/2031af314e70f3b8e07add13cb65416c29956181/ql/src/test/org/apache/hadoop/hive/ql/txn/compactor/TestWorker.java#L1190
Which simulates the timeout scenarios for worker and when worker goes in timeout we send the thread to sleep for long duration of time. To avoid sending threads to go to sleep for long duration while testing we set lower values for these in tests.

Reply:
ok 👍